### PR TITLE
There was a spurious trigger to spin up the spray machinery, rewire

### PR DESCRIPTION
**Pull request by geoffjentry**

### CODE REVIEW

**Comment by geoffjentry** [comment]
Note that this does undo PR #340 - that one ended up fixing a symptom, not the underlying issue

File: src/main/scala/cromwell/server/CromwellServer.scala [hunk 1]
```diff
@@ -2,6 +2,7 @@ package cromwell.server
 
 import akka.util.Timeout
 import com.typesafe.config.ConfigFactory
+import cromwell.engine.backend.CromwellBackend
```
Comment (mcovarr) [inline src/main/scala/cromwell/server/CromwellServer.scala]: unused

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala [hunk 2]
```diff
@@ -9,13 +9,12 @@ import ch.qos.logback.core.FileAppender
...
 import cromwell.binding.values.WdlFile
-import cromwell.engine.backend.Backend
+import cromwell.engine.backend.{CromwellBackend, Backend}
 import cromwell.engine.backend.jes.JesBackend
 import cromwell.engine.io.IoManager
 import cromwell.engine.io.gcs.GoogleCloudStorage
 import cromwell.engine.io.shared.SharedFileSystemIoInterface
 import cromwell.engine.workflow.WorkflowOptions
-import cromwell.server.CromwellServer
 import cromwell.util.TryUtil
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: unused stuff here too

File: src/main/scala/cromwell/Main.scala [hunk 3]
```diff
@@ -80,7 +80,7 @@ class Main private[cromwell](enableTermination: Boolean, managerSystem: () => Wo
   lazy val Log = LoggerFactory.getLogger("cromwell")
   Monitor.start()
 
-  def this() = this(enableTermination = true, managerSystem = () => CromwellServer)
+  def this() = this(enableTermination = true, managerSystem = () => new WorkflowManagerSystem {})
 
   // CromwellServer still doesn't clean up... so => Any
   def runAction(args: Seq[String]): Any = {
```
Comment (mcovarr) [inline src/main/scala/cromwell/Main.scala]: `parseInputs` is no longer used in this file.  Is that a result of your previous de-redundification?

**Comment by mcovarr** [comment]
:+1:

File: src/main/scala/cromwell/server/CromwellServer.scala [hunk 4]
```diff
@@ -9,13 +10,12 @@ import scala.concurrent.duration._
...
-  val conf = ConfigFactory.load()
-  val service = actorSystem.actorOf(CromwellApiServiceActor.props(workflowManagerActor, conf), "cromwell-service")
   implicit val timeout = Timeout(5.seconds)
-  val webserviceConf = conf.getConfig("webservice")
-
   import scala.concurrent.ExecutionContext.Implicits.global
 
+  val conf = ConfigFactory.load()
+  val service = actorSystem.actorOf(CromwellApiServiceActor.props(workflowManagerActor, conf), "cromwell-service")
+  val webserviceConf = conf.getConfig("webservice")
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/server/CromwellServer.scala]: This looks the same - is it just an aesthetic rearrangement?

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala [hunk 5]
```diff
@@ -99,7 +98,7 @@ object WorkflowDescriptor {
   }
 
   def apply(id: WorkflowId, sourceFiles: WorkflowSourceFiles, conf: Config): WorkflowDescriptor = {
-    validateWorkflowDescriptor(id, sourceFiles, CromwellServer.backend, conf) match {
+    validateWorkflowDescriptor(id, sourceFiles, CromwellBackend.backend(), conf) match {
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: Any reason for the choice of adding in ()? I thought they were considered thoughtcrime in our new, enlightened age

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala (hunk 2 above)
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: But is it worth mentioning that IntelliJ has no idea what's going on in the middle of this file? I don't know how far I'd trust it's auto-wizardry here

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala [hunk 6]
```diff
@@ -99,7 +98,7 @@ object WorkflowDescriptor {
   }
 
   def apply(id: WorkflowId, sourceFiles: WorkflowSourceFiles, conf: Config): WorkflowDescriptor = {
-    validateWorkflowDescriptor(id, sourceFiles, CromwellServer.backend, conf) match {
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: I noticed that WorkflowManagerSystem (extended by CromwellServer) still defines "backend" but it's only ever used by tests. And I'm pretty sure they aren't *really* using it.

**Comment by cjllanwarne** [comment]
Other than my minigripes, :+1:

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala (hunk 6 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: That's not *completely* true. The object's construction of WMA references it as well. This all stems from a change I'm not particularly proud of but was the lesser of two hopefully temporary evils a while back

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala (hunk 5 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: They're thoughtcrime except when it is thoughtcrime to *not* do it (aka intellij reminded me)

File: src/main/scala/cromwell/Main.scala (hunk 3 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/Main.scala]: Pretty sure that got lost in one of y mental rebases

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala (hunk 2 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: That's actually the reason you all catch so many import shenanigans from me. From older versions of intellij I grew a healthy distrust of its auto-import-doo-dad for scala and i often forget to do it by hand

File: src/main/scala/cromwell/server/CromwellServer.scala (hunk 4 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/server/CromwellServer.scala]: not originally but yes

**Pull request merged**